test/externalTests/plugins/testCommon.js

```diff
@@ -215,60 +204,73 @@ function inject (bot) {
+    return new Promise(resolve => {
+      const onChatMessage = (username, message) => {
+        if (username === to && listen(message)) {
+          bot.removeListener('chat', onChatMessage)
+          resolve()
+        }
       }
-    }
-    bot.on('chat', chatHandler)
-    bot.chat(what)
+      bot.on('chat', onChatMessage)
+      bot.chat(what)
+    })
   }
 
-  function runExample (file, run, cb) {
-    let childBotName
-    function joinHandler (message) {
-      if (message.json.translate === 'multiplayer.player.joined') {
-        bot.removeListener('message', joinHandler)
-        childBotName = message.json.with[0].insertion
-        bot.chat(`/tp ${childBotName} 50 4 0`)
-        setTimeout(() => {
-          bot.chat('loaded')
-        }, 5000)
+  async function runExample (file, run) {
+    return new Promise((resolve, reject) => {
+      let childBotName
+
+      function joinHandler (message) {
+        if (message.json.translate === 'multiplayer.player.joined') {
+          bot.removeListener('message', joinHandler)
+          childBotName = message.json.with[0].insertion
+          bot.chat(`/tp ${childBotName} 50 4 0`)
+          setTimeout(() => {
+            bot.chat('loaded')
+          }, 5000)
+        }
       }
-    }
-    bot.on('chat', (username, message) => {
-      if (message === 'Ready!') {
-        run(childBotName, closeExample)
+
+      const onChatMessage = (username, message) => {
+        if (message === 'Ready!') {
+          run(childBotName, closeExample)
+        }
       }
-    })
-    bot.on('message', joinHandler)
+      bot.on('chat', onChatMessage)
+      bot.on('message', joinHandler)
 
-    const child = spawn('node', [file, 'localhost', `${bot.test.port}`])
+      const child = spawn('node', [file, 'localhost', `${bot.test.port}`])
 
-    // Useful to debug child processes:
-    child.stdout.on('data', (data) => { console.log(`${data}`) })
-    child.stderr.on('data', (data) => { console.error(`${data}`) })
+      // Useful to debug child processes:
+      child.stdout.on('data', (data) => { console.log(`${data}`) })
+      child.stderr.on('data', (data) => { console.error(`${data}`) })
 
-    const timeout = setTimeout(() => {
-      console.log('Timeout, test took too long')
-      closeExample(new Error('Timeout, test took too long'))
-    }, 20000)
+      const timeout = setTimeout(() => {
+        console.log('Timeout, test took too long')
+        closeExample(new Error('Timeout, test took too long'))
+      }, 20000)
 
-    function closeExample (err) {
-      if (timeout) clearTimeout(timeout)
-      console.log('kill process ' + child.pid)
+      function closeExample (err) {
+        if (timeout) clearTimeout(timeout)
+        console.log('kill process ' + child.pid)
 
-      child.once('close', (code) => {
-        console.log('close requested ' + code)
-        cb(err)
-      })
-      process.kill(child.pid, 'SIGTERM')
-    }
+        bot.removeListener('chat', onChatMessage)
```

Comment:
Listener was previously not being removed, and I was noticing issues with listener limits being hit in the test output